### PR TITLE
Refactor the Duplicated code in OTLP Exporter with config part

### PR DESCRIPTION
**Pull request by hanyuancheung**
Fix: part of #2641 
- [x] Refactor otlpconfig part from `exporter/otlp/otlpmetric(otlptrace)/internal/` into `exporter/otlp/internal/envconfig`
- [ ] Refactor transform part from `exporter/otlp/otlpmetric(otlptrace)/internal/` into `exporter/otlp/internal/transform`
- [ ] Refactor connection part from `exporter/otlp/otlpmetric(otlptrace)/internal/` `exporter/otlp/internal/connection`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,9 @@ require (
 	github.com/go-logr/stdr v1.2.2
 	github.com/google/go-cmp v0.5.7
 	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/otel/exporters/otlp/internal/retry v0.0.0-00010101000000-000000000000
 	go.opentelemetry.io/otel/trace v1.5.0
+	google.golang.org/grpc v1.45.0
```
Comment (MadVikingGod) [inline go.mod:12]: This change will add the otlp internals to every project that depends on the otel API.  Which is why every example and bridge just gained 40-50 lines of dependences in the go.sum files.

Comment (hanyuancheung) [reply go.mod:12]: I agree, but I didn't notice when this dependency was introduced. Do you have some suggestions to resolve this problem? @MadVikingGod

Comment (MadVikingGod) [reply go.mod:12]: My guess is that `exporters/otlp/internal` is under the main otlp mod (`/go.mod`) so when you moved these there you brought with it the grpc etc dependencies.

To house it there you will need a mod somewhere, either /exporter/otlp, /exporters/otlp/internal or /exporters/otlp/internal/pkg.  I would experiment what works best and also remember to add it to the versions.yaml and run `make crosslink` or `make precommit`.

**Comment by hanyuancheung** [comment]
Tnanks for advice, I've added the go.mod in `exporters/otlp/internal`, it seems to look good. cc @MadVikingGod

**Review by MadVikingGod — APPROVED** [review]
Other then the Changelog LGTM

**Pull request closed without merging**